CMakePresets.json

```diff
@@ -1,99 +1,251 @@
 {
-  "version": 2,
+  "version": 3,
   "configurePresets": [
     {
-      "name": "linux-clang-debug",
-      "binaryDir": "${sourceDir}/build",
-      "generator": "Unix Makefiles",
+      "name": "windows-base",
```

Comment:
It might be good to rename this to `windows-msvc-base` to distinguish it from `windows-clang-base`

Reply:
The idea here was to treat MSVC as the default for Windows, the same way as GCC is the default for Linux. The Clang variants inherit all settings but change the compiler. 

We could change `linux-base` -> `linux-gcc-base` too in that case. What do you think?

Reply:
I like having `linux-base` separate from `linux-gcc-base` and `linux-clang-base` because the "base" shouldn't be tied to a vendor in my opinion. Plus it potentially reduces the risk of somebody adding something vendor-specific to the base preset and forgetting to either override or disinherit it in the other preset in the future.

Reply:
Makes sense - that's done now, also removing the generators from the base configs.